exchanges/huobi/huobi_wrapper.go

```diff
@@ -149,76 +149,87 @@ func (h *HUOBI) UpdateOrderbook(p pair.CurrencyPair, assetType string) (orderboo
 	return orderbook.GetOrderbook(h.Name, p, assetType)
 }
 
-var mtx sync.Mutex
-
-// GetAccountID returns the account ID for trades NOTE interim implementation
-// does not account for multiple account IDs
-func (h *HUOBI) GetAccountID() (string, error) {
-	mtx.Lock()
-	defer mtx.Unlock()
-
-	if h.AccountID == "" {
-		acc, err := h.GetAccounts()
-		if err != nil {
-			return "", err
-		}
-
-		if len(acc) > 0 {
-			return strconv.FormatInt(acc[0].ID, 10), nil
-		}
+// GetAccountID returns the account ID for trades
+func (h *HUOBI) GetAccountID() ([]Account, error) {
+	acc, err := h.GetAccounts()
+	if err != nil {
+		return nil, err
+	}
 
-		return "", errors.New("no user ID fetched")
+	if len(acc) < 1 {
+		return nil, errors.New("no account returned")
 	}
 
-	return h.AccountID, nil
+	return acc, nil
 }
 
 //GetAccountInfo retrieves balances for all enabled currencies for the
 // HUOBI exchange - to-do
 func (h *HUOBI) GetAccountInfo() (exchange.AccountInfo, error) {
 	var info exchange.AccountInfo
-	info.ExchangeName = h.GetName()
+	info.Exchange = h.GetName()
 
-	accID, err := h.GetAccountID()
+	accounts, err := h.GetAccountID()
 	if err != nil {
 		return info, err
 	}
 
-	acc, err := h.GetAccountBalance(accID)
-	if err != nil {
-		return info, err
-	}
+	for _, account := range accounts {
+		var acc exchange.Account
 
-	type hold struct {
-		Avail float64
-		Hold  float64
-	}
+		acc.ID = strconv.FormatInt(account.ID, 10)
 
-	var currencyData = make(map[string]*hold)
-	for _, data := range acc {
-		_, ok := currencyData[data.Currency]
-		if !ok {
-			currencyData[data.Currency] = &hold{}
+		if account.State == "working" {
+			acc.Working = true
```

Comment:
This is the only place I see `.Working` having the ability to be `false`. However, I don't see anything that changes behaviour for when the account is false. Would you be able to explain what the working is used for?